internal/store/engine_state.go

```diff
@@ -112,6 +112,23 @@ type Pod struct {
 	OldRestarts       int // # times the pod restarted when it was running old code
 }
 
+// attempting to include the most recent crash, but no preceding crashes
+// (e.g., we don't want to show the same panic 20x in a crash loop)
+// if the current pod has crashed, then just print the current pod
+// if the current pod is live, print the current pod plus the last pod
+func (p Pod) Log() string {
+	var podLog string
+	// if the most recent pod is up, then we want the log from the last run (if any), since it crashed
+	if p.ContainerReady {
+		podLog = string(p.PreRestartLog) + string(p.CurrentLog)
+	} else {
+		// otherwise, the most recent pod has the crash itself, so just return itself
+		podLog = string(append([]byte{}, p.CurrentLog...))
```

Comment:
is there a reason you copy this twice? note that the `string()` call already makes a copy

Reply:
I'd been thinking of `string()` as a cast, and apparently had not yet felt the need to look up whether go strings are immutable. Thanks!